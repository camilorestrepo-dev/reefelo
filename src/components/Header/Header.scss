@import "../../styles/shared.scss";

.navbar {
  position: sticky;
  top: 0;
  width: 100%;
  background-color: $primary;
  color: white;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
  z-index: 1000;

  .navbar-container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: $spacing-12 $spacing-16;
    max-width: $max-page-width;
    margin: 0 auto;

    .navbar-logo {
      font-size: 1.5rem;
    }

    .navbar-links {
      list-style: none;
      display: flex;
      gap: 1rem;
      margin: 0;
      padding: 0;

      .logout-btn {
        font-weight: normal;
        color: $text-color;
        border-color: $text-color;
      }
    }
  }
}