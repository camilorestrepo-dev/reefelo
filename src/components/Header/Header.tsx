import { useTranslation } from "react-i18next";
import Button from "../Button/Button";
import { useAuth } from "../../hooks/useAuth";
import { NavLink } from "react-router-dom";

import "./Header.scss";

const Header: React.FC = () => {
  const { t } = useTranslation();
  const { logout } = useAuth();

  return (
    <header className="navbar">
      <nav className="navbar-container">
        <NavLink to="/home" className="navbar-logo">
          Reefelo
        </NavLink>
        <ul className="navbar-links">
          <li>
            <Button variant="outline" className="logout-btn" onClick={logout}>
              {t("signOut")}
            </Button>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
