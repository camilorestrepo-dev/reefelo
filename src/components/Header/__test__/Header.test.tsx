import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { I18nextProvider } from "react-i18next";
import Header from "../Header";
import i18n from "../../../locale";

// Mock the custom hook `useAuth`
jest.mock("../../../hooks/useAuth", () => ({
  useAuth: jest.fn(() => ({
    signInError: "",
    signInLoading: false,
    clearSignInError: jest.fn(),
    login: jest.fn(),
    logout: jest.fn(),
  })),
}));

// Render providing a Browser Router context
const renderWithRouter = (ui: React.ReactNode) => {
  window.history.pushState({}, "Test page", "/");
  return render(ui, { wrapper: BrowserRouter });
};

test("render correctly", () => {
  const component = renderWithRouter(
    // Render within an i18n context
    <I18nextProvider i18n={i18n}>
      <Header />
    </I18nextProvider>,
  );
  expect(component.container).toMatchSnapshot();
});
