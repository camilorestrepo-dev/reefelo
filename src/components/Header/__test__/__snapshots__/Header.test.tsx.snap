// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div>
  <header
    class="navbar"
  >
    <nav
      class="navbar-container"
    >
      <a
        class="navbar-logo"
        href="/home"
      >
        Reefelo
      </a>
      <ul
        class="navbar-links"
      >
        <li>
          <button
            class="custom-btn outline-btn logout-btn"
            type="button"
          >
            Sign Out
          </button>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
